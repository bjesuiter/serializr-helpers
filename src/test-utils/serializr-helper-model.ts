import {list, primitive, serializable} from "serializr";

export const demoProp: string = 'Bob der Baumeister';

export const demoArray: string[] = [
    'Lilly',
    'Gunnar',
    'Fjordojan'
];

export class SerializrHelperModel {

    @serializable(primitive())
    demoProp: string = demoProp;

    @serializable(list(primitive()))
    demoArray: string[] = demoArray;

}
