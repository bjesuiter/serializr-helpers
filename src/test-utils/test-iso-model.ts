import {serializable} from "serializr";
import {Moment} from "moment";
import {MomentSerializationSchema} from "../lib/moment-serialization-schema";

export class TestIsoModel {

    @serializable(MomentSerializationSchema())
    public testMoment: Moment;

    constructor(testMoment: Moment) {
        this.testMoment = testMoment;
    }


}
