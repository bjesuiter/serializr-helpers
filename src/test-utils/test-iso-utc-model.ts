import {serializable} from "serializr";
import {Moment} from "moment";
import {MomentSerializationSchema} from "../lib/moment-serialization-schema";

export class TestIsoUtcModel {

    @serializable(MomentSerializationSchema({useUtc: true}))
    public testMoment: Moment;

    constructor(testMoment: Moment) {
        this.testMoment = testMoment;
    }


}
