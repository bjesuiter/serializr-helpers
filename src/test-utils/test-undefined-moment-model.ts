import {serializable} from "serializr";
import {Moment} from "moment";
import {MomentSerializationSchema} from "../lib/moment-serialization-schema";

export class TestUndefinedMomentModel {

    @serializable(MomentSerializationSchema())
    public testMoment?: Moment;

    constructor(testMoment?: Moment) {
        this.testMoment = testMoment;
    }
}
