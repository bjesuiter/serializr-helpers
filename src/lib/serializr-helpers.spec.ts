import {SerializrHelperModel} from "../test-utils/serializr-helper-model";
import {
    deserializeFromJson,
    deserializeFromPojo,
    deserializeJson,
    serializeJson,
    serializeToJson,
    serializeToPojo
} from "./serializr-helpers";
import {deserialize, serialize} from "serializr";

describe('Serializr Helpers', () => {

    it('serialize to JSON', () => {
        const helper = new SerializrHelperModel();
        const targetJson = serializeToJson(helper);
        const validateJson = JSON.stringify(serialize(helper));

        expect(targetJson).toMatchSnapshot();
        expect(targetJson).toEqual(validateJson);
    });

    it('deserialize from JSON', () => {
        const helperModel = new SerializrHelperModel();
        const helperJson = serializeToJson(helperModel);
        const targetModel = deserializeFromJson(SerializrHelperModel, helperJson);
        const validateModel = deserialize(SerializrHelperModel, JSON.parse(helperJson));

        expect(targetModel).toEqual(validateModel);
        expect(targetModel).toBeInstanceOf(SerializrHelperModel);
        expect(validateModel).toBeInstanceOf(SerializrHelperModel);
    });

    it('serialize JSON alias for serializeToJson', () => {
        const helper = new SerializrHelperModel();
        const targetJson = serializeJson(helper);
        const validateJson = JSON.stringify(serialize(helper));

        expect(targetJson).toMatchSnapshot();
        expect(targetJson).toEqual(validateJson);
    });

    it('deserialize JSON alias for deserializeFromJson', () => {
        const helperModel = new SerializrHelperModel();
        const helperJson = serializeJson(helperModel);
        const targetModel = deserializeJson(SerializrHelperModel, helperJson);
        const validateModel = deserialize(SerializrHelperModel, JSON.parse(helperJson));

        expect(targetModel).toEqual(validateModel);
        expect(targetModel).toBeInstanceOf(SerializrHelperModel);
        expect(validateModel).toBeInstanceOf(SerializrHelperModel);
    });

    it('serialize To Pojo', () => {
        const helperModel = new SerializrHelperModel();
        const targetPojo = serializeToPojo(helperModel);
        const validatePojo = serialize(helperModel);

        expect(targetPojo).toEqual(validatePojo);
    });

    it('deserialize from Pojo', () => {
        const helperModel = new SerializrHelperModel();
        const helperPojo = serialize(helperModel);
        const targetModel = deserializeFromPojo(SerializrHelperModel, helperPojo);
        const validateModel = deserialize(SerializrHelperModel, helperPojo);

        expect(targetModel).toEqual(validateModel);
        expect(targetModel).toBeInstanceOf(SerializrHelperModel);
        expect(validateModel).toBeInstanceOf(SerializrHelperModel);
    });
});
