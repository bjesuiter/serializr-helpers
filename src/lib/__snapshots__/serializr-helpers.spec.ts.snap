// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Serializr Helpers serialize JSON alias for serializeToJson 1`] = `"{\\"demoMoment\\":\\"2019-06-10T08:50:23.559+02:00\\",\\"demoProp\\":\\"Bob der Baumeister\\",\\"demoArray\\":[\\"Lilly\\",\\"Gunnar\\",\\"Fjordojan\\"]}"`;

exports[`Serializr Helpers serialize to JSON 1`] = `"{\\"demoMoment\\":\\"2019-06-10T08:50:23.559+02:00\\",\\"demoProp\\":\\"Bob der Baumeister\\",\\"demoArray\\":[\\"Lilly\\",\\"Gunnar\\",\\"Fjordojan\\"]}"`;
